### PR TITLE
master-qa-50756

### PR DESCRIPTION
**Pull request by chaowang-123**
https://issues.liferay.com/browse/LRQA-50756
Submit to master.

### CODE REVIEW

**Comment by chaowang-123** [comment]
ci:test:sf

**Comment by Sharryxue** [comment]
Just starting review.

File: portal-web/test/functional/com/liferay/portalweb/macros/GDPR.macro [hunk 1]
```diff
@@ -146,7 +146,7 @@ definition {
 		Alert.viewSuccessMessage();
 	}
 
-	macro anonymizeStagedDatainApplicationCP {
+	macro anonymizeStagedAndLiveDatainApplicationCP {
```
Comment (Sharryxue) [inline portal-web/test/functional/com/liferay/portalweb/macros/GDPR.macro]: rename it to anonymizeTheSpecificEntry

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/administration/users/gdpr/GDPR.testcase [hunk 2]
```diff
@@ -455,13 +455,93 @@ definition {
 		}
 	}
 
-	@description = "This test anonymizes a staged and a live entry from an applciation."
-	@ignore = "Test Stub"
+	@description = "This test anonymizes a staged and a live entry from an applciation, test with WC entries."
 	@priority = "3"
-	test CanAnonymizeASingleStagedAndLiveEntryFromAnApplication {
+	test CanAnonymizeASingleStagedAndLiveEntryFromWebContent {
```
Comment (Sharryxue) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/administration/users/gdpr/GDPR.testcase]: Rename it to CanAnonymizeASingleStagedAndLiveWebContentEntry

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/administration/users/gdpr/GDPR.testcase [hunk 3]
```diff
@@ -455,13 +455,93 @@ definition {
...
+				layoutName = "Staging Test Page");
+
+			JSONUser.addUserToSite(
+				groupName = "Site Name",
+				userEmailAddress = "userea@liferay.com");
+
+			JSONStaging.enableLocalStaging(groupName = "Site Name");
+		}
+
+		task ("Given add the staged entry with web content and publish to live this entry") {
```
Comment (Sharryxue) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/administration/users/gdpr/GDPR.testcase]: We not using multi Given when apply the BDD format, we can either put them in one given block or nested by using And.
In order to put the Given together, we can use a description like this e.g.:Publish the staged web content entries to live as a new admin user.

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/administration/users/gdpr/GDPR.testcase [hunk 4]
```diff
@@ -455,13 +455,93 @@ definition {
...
+				userLoginFullName = "Test Test");
 
+			User.openUsersAdmin();
+
+			User.deletePersonalDataCP(
+				userFirstName = "userfn",
+				userLastName = "userln");
+		}
+
+		task ("And select anonymize staged 'GDPR Anonymize Article 1' entry and lived 'GDPR Anonymize Article 2' entry") {
```
Comment (Sharryxue) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/administration/users/gdpr/GDPR.testcase:509]: If it's hard to combine the When, using And to nested it like following.
```
task ("When balbalbalbla")
    task ("And balbalbalbla")
```

**Review by Sharryxue — CHANGES_REQUESTED** [review]
Hello @chaowang-123 
Thanks for applying the BDD format, there are some changes required before forwarding it.

Bests

**Comment by Sharryxue** [comment]
Just starting review.

**Comment by Sharryxue** [comment]
Made following changes.
- Squashed the commits, when we have need update the changes, instead of adding an additional commit, modify the exist commits that have issues.
- Combined the Given task
- Nested the And block into when task
- Shorten the commit message
View [3dc5527%5E...a164e94](https://github.com/Sharryxue/liferay-portal/compare/3dc5527%5E...a164e94)

Resend [Sharryxue/liferay-portal#371](https://github.com/Sharryxue/liferay-portal/pull/371)

**Pull request closed without merging**